src/components/PostEntry.js

```diff
@@ -1,15 +1,63 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { v4 } from 'uuid';
+
+// types
+import { PostType } from '../types/post.type';
+import { UserType } from '../types/user.type';
+
+// components
+import CommentInput from './CommentInput';
+import CommentList from './CommentList';
 
 const PostEntry = ({ post, user }) => {
-    const { title, body } = post || {};
-    const { name } = user || {};
+    const { title, body } = post;
+    const { name } = user;
+
+    const [likeCounter, setLikeCounter] = useState(0);
+    const [theme, setTheme] = useState('light');
+    const [comments, setComments] = useState([]);
+
+    const updateCounter = () => {
+        setLikeCounter(previousCounter => previousCounter + 1);
+    };
+    const updateTheme = selectedTheme => {
+        setTheme(selectedTheme);
+    };
+    const addComment = comment => {
+        setComments(prev => [...prev, { uuid: v4(), ...comment }]);
+    };
```

Comment:
Do you think we could add docblock with @see comment pointing to uuid docs, and maybe explain what the code do where we using it? I hope this would exposed students to both `uuid` and JSDoc blocks, which would expand their knowledege outside our curriculum (not only folowing code example).

```suggestion
    /*
     * Add new comment with unique identifier
     *
     * @see https://www.npmjs.com/package/uuid
    */
    const addComment = comment => {
        setComments(prev => [...prev, { uuid: v4(), ...comment }]);
    };
```

Reply:
addressed in 400afbd021662b78a63fedc579067141e8d3f437